infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInternalRuntime.java

```diff
@@ -200,6 +204,24 @@ private void prepareOpenShiftProject(String projectName) throws InfrastructureEx
         }
     }
 
+    private void prepareOpenShiftPVCs(OpenShiftEnvironment osEnv, String namespace) throws InfrastructureException {
+        try (OpenShiftClient client = clientFactory.create()) {
+            final Map<String, PersistentVolumeClaim> existing = new HashMap<>();
+            client.persistentVolumeClaims()
+                  .inNamespace(namespace)
+                  .list()
+                  .getItems()
+                  .forEach(p -> existing.put(p.getMetadata().getName(), p));
+            final Map<String, PersistentVolumeClaim> toCreate =
+                    Maps.difference(osEnv.getPersistentVolumeClaims(), existing).entriesOnlyOnLeft();
+            for (PersistentVolumeClaim pvc : toCreate.values()) {
```

Comment:
here you're using old (previous JDK8) for each loop, is it wanted .
also, could we filtered directly in the stream ?